### PR TITLE
Add defines for byond-tracy support

### PR DESCRIPTION
**Pull request by Mothblocks**
Adds `USE_BYOND_TRACY`, which automatically loads a given prof.dll/libprof.so using https://github.com/mafemergency/byond-tracy/.

Not intended for use in production, and we do not ship a copy of byond-tracy. It is extremely easy to compile yourself, but if you're someone interesting enough to play around with this then let me know and I can just give you a build. 

I'm going to be using this for init profiling.
Tracy_2022-10-30T12-51-02

### CODE REVIEW

File: code/game/world.dm [hunk 1]
```diff
@@ -385,6 +389,22 @@ GLOBAL_VAR(restart_counter)
...
+
+	switch (system_type)
+		if (MS_WINDOWS)
+			library = "prof.dll"
+		if (UNIX)
+			library = "libprof.so"
+		else
+			CRASH("Unsupported platform: [system_type]")
+
+	var/init_result = call(library, "init")()
```
Comment (Mothblocks) [inline code/game/world.dm:403]: @AnturK Gonna need to be call_ext if your 515 branch gets merged before this PR

Comment (AnturK) [reply code/game/world.dm:403]: Mhm

Comment (Mothblocks) [reply code/game/world.dm:403]: I meant my comment the other way around but you know what I mean

File: code/game/world.dm [hunk 2]
```diff
@@ -385,6 +389,22 @@ GLOBAL_VAR(restart_counter)
...
+/world/proc/init_byond_tracy()
+	var/library
+
+	switch (system_type)
+		if (MS_WINDOWS)
+			library = "prof.dll"
+		if (UNIX)
+			library = "libprof.so"
+		else
+			CRASH("Unsupported platform: [system_type]")
```
Comment (AffectedArc07) [inline code/game/world.dm:401]: Realistically is this else branch needed?

Comment (Mothblocks) [reply code/game/world.dm:401]: Preference thing. It's unused in practice (and it's unlikely Lummox supports Mac or anything else) but in principle stuff like system type should be treated as a non-exhaustive enum IMO

**Comment by ShizCalev** [comment]
I'd love a build!

File: code/game/world.dm (hunk 2 above)
Comment (Iamgoofball) [reply code/game/world.dm:401]: you can't forget about the Plan 9 gamers, @AffectedArc07

Comment (AffectedArc07) [reply code/game/world.dm:401]: > you can't forget about the Plan 9 gamers, @AffectedArc07 

They don't deserve this luxury

**Pull request merged**